src/expr/src/scalar/mod.rs

```diff
@@ -306,9 +306,15 @@ impl MirScalarExpr {
                 if expr.is_literal() {
                     *e = eval(e);
                 } else if *func == UnaryFunc::IsNull {
-                    // (<expr1> <op> <expr2>) IS NULL can often be simplified to
-                    // (<expr1> IS NULL) OR (<expr2> IS NULL).
-                    if let MirScalarExpr::CallBinary { func, expr1, expr2 } = &mut **expr {
+                    if let MirScalarExpr::Column(c) = &**expr {
```

Comment:
This branch gas an extra ```isnull``` that is not present with the previous commit:

```
materialize=>  EXPLAIN  SELECT  FROM t2  AS a1  JOIN t2  AS a2  ON ( a1 .f2  > a1 .f2  )  WHERE a2 .f1  + a1 .f1  + a1 .f2  IS  NOT  NULL  AND a2 .f1  IS  NULL  AND  7  = a1 .f1   ;
               Optimized Plan                
---------------------------------------------
 %0 =                                       +
 | Get materialize.public.t2 (u13)          +
 | Filter !(isnull(#0)), (#0 = 7), (#1 > #1)+ <- This isnull
 | ArrangeBy ()                             +
                                            +
 %1 =                                       +
 | Get materialize.public.t2 (u13)          +
 | Filter !(isnull(#0)), isnull(#0)         +
                                            +
 %2 =                                       +
 | Join %0 %1                               +
 | | implementation = Differential %1 %0.() +
 | | demand = ()                            +
 | Project ()                               +
```

This seems to depend quasi-randomly on the complexity of the expression, that is, the main branch will also have this ```isnull``` for certain expressions, just not this one.